arch/risc-v/src/common/up_initialize.c

```diff
@@ -113,6 +116,13 @@ void up_initialize(void)
 
   up_addregion();
 
+  /* Initialize the idle task stack info */
+
+  idle = this_task(); /* It should be idle task */
+  idle->stack_alloc_ptr = _END_BSS;
+  idle->adj_stack_ptr   = (FAR void *)g_idle_topstack;
+  idle->adj_stack_size  = CONFIG_IDLETHREAD_STACKSIZE;
+
```

Comment:
It would be nice to generalize this for all architectures.  There is description in top-level TODO list of how this might be done in nx_start().

Putting this logic in each up_initialize() is probably fine too since we really cannot handle faults until up_initailize() is called.

Reply:
Sure, I will provide a patch to apply the similar change to other arch tomorrow.

Reply:
For most architectures, the adj_stack_size might not be CONFIG_IDLETHREAD_STACKSIZE.  it might be smaller to alignment requirements.

Reply:
TizenRT implemented this by forcing all architectures to following the same pattern: https://github.com/Samsung/TizenRT/blob/master/os/kernel/init/os_start.c#L369

But they do not support very man architectures.  For NuttX, most but not all architectures follow this pattern as described in the TODO list.